### ISSUE TITLE
🐛 Bug: One-To-One Link Field - Cannot Search

### ISSUE THREAD

**Issue by Nick-PDS** [nocodb/nocodb#9439]
### Please confirm that the bug report does ***not*** already exist

- [X] I confirm there is no existing issue for this bug.

### Steps to reproduce

When attempting to search in the popup for a record to link, I keep getting a "Failed to load list: Something went wrong". Diving into the network requests this is the backend error I get from the API. This exclusively only happens to fields that are a One-To-One link, when I use "Has Many" the search bar in the popup works.

Error from the API:
{"msg":"The operator \"undefined\" is not permitted"}

API Endpoint throwing error:
/api/v1/db/data/noco/px6xbw61wsmahds/m2vqq0loky8uohg/1/oo/cki8f7528tbw6v2/exclude?limit=10&offset=0&where=(First+Name,like,a)&linkRowData=%7B%7D

1. Create Table A with fields
2. Create Table B with fields
3. Add a One-To-One Link field in TABLE A
4. Go back to data on TABLE A and attempt search for record to link

### Desired Behavior

Search bar returns successful results

### Project Details

Node: **v20.15.1**
Arch: **x64**
Platform: **linux**
Docker: **true**
RootDB: **pg**
PackageVersion: **0.255.2**

### Attachments

https://github.com/user-attachments/assets/0e52e9d2-29dd-4afc-aad8-bf42145e687f

**Comment by ionws** [comment]
+1 to this bug.

**Comment by dstala** [comment]
@ionws let us know if the issue is reproducible still. i tried in latest release & worked fine.
@Nick-PDS

**Comment by dstala** [comment]
Please reopen this ticket if issue is reproducible still

**Comment by Omasumasu** [comment]
I think it's still happened with v0.258.2

**Comment by pranavxc** [comment]
> I think it's still happened with v0.258.2

@Omasumasu 

Not able to reproduce locally with latest docker image. Please share more info,

- Error you are getting in UI ?
- Database you are using ?
- Backend error log(docker) if any.

It will be better if you can record a video and share.

**Comment by Omasumasu** [comment]
@pranavxc 

>

> > I think it's still happened with v0.258.2
> 
> @Omasumasu
> 
> Not able to reproduce locally with latest docker image. Please share more info,
> 
> * Error you are getting in UI ?
> * Database you are using ?
> * Backend error log(docker) if any.
> 
> It will be better if you can record a video and share.

I got error message `Something went wrong` when typed search box.

And in Chrome development tool, network tabs shows request error when request `https://xxx.com/api/v1/db/data/noco/pzc3c5q5v13ioiz/meop3e52vjafy4e/1/oo/c74gcagxj210zqn/exclude?limit=10&offset=0&where=(XXXXX)&linkRowData=XXXXX`

```
{"msg":"The operator \"undefined\" is not permitted"}
```

We are using PostgreSQL 16 on RDS.

Found error logs on docker.

```
2024-11-27T08:30:31.533Z
TypeError: The operator "undefined" is not permitted
2024-11-27T08:30:31.533Z
at operator (/usr/src/app/node_modules/knex/lib/formatter/wrappingFormatter.js:132:11)
2024-11-27T08:30:31.533Z
at QueryCompiler_PG.whereBasic (/usr/src/app/node_modules/knex/lib/query/querycompiler.js:1057:7)
2024-11-27T08:30:31.533Z
at QueryCompiler_PG.where (/usr/src/app/node_modules/knex/lib/query/querycompiler.js:573:34)
2024-11-27T08:30:31.533Z
at QueryCompiler_PG.toSQL (/usr/src/app/node_modules/knex/lib/query/querycompiler.js:73:29)
2024-11-27T08:30:31.533Z
at compileCallback (/usr/src/app/node_modules/knex/lib/formatter/formatterUtils.js:13:19)
2024-11-27T08:30:31.533Z
at rawOrFn (/usr/src/app/node_modules/knex/lib/formatter/wrappingFormatter.js:225:7)
2024-11-27T08:30:31.533Z
at QueryCompiler_PG.whereWrapped (/usr/src/app/node_modules/knex/lib/query/querycompiler.js:1084:17)
2024-11-27T08:30:31.533Z
at QueryCompiler_PG.where (/usr/src/app/node_modules/knex/lib/query/querycompiler.js:573:34)
2024-11-27T08:30:31.533Z
at QueryCompiler_PG.toSQL (/usr/src/app/node_modules/knex/lib/query/querycompiler.js:73:29)
2024-11-27T08:30:31.533Z
at compileCallback (/usr/src/app/node_modules/knex/lib/formatter/formatterUtils.js:13:19)
```

I can't show you video capture because it contains privacy data.

**Comment by Omasumasu** [comment]
@pranavxc 

```&where=(%E5%A5%91%E7%B4%84%E6%9D%A1%E9%A0%85%E5%90%8D,like,xx)``

This field use multibyte character, but I think it's not root cause.
Other columns with 1 to 1 relation  with multibyte character works correctly.
But I can't find difference of those.

**Comment by Omasumasu** [comment]
@pranavxc 

One possible cause may be that another column in the same table has a one-to-one relationship with the same table.

**Comment by pranavxc** [comment]
> @pranavxc
> 
> ```&where=(%E5%A5%91%E7%B4%84%E6%9D%A1%E9%A0%85%E5%90%8D,like,xx)``
> 
> This field use multibyte character, but I think it's not root cause. Other columns with 1 to 1 relation with multibyte character works correctly. But I can't find difference of those.

I think that causing the issue if you can rename and try once we can confirm it as well. Let me try it locally as well.

**Comment by Omasumasu** [comment]
@pranavxc 
Renaming didn't solve the problem😢

**Comment by pranavxc** [comment]
> @pranavxc Renaming didn't solve the problem😢

I've couple of scenarios in my mind, will test all those and get back to you.

**Comment by pranavxc** [comment]
Reproducible when I set display value column name of target table with the value you provided ( `契約条項名` ).

**Comment by pranavxc** [comment]
@Omasumasu  It was not related to column name, the real issue happened because it's trying to extract column details from wrong table. We made a fix here - https://github.com/nocodb/nocodb/pull/9903

You can verify the fix by using following PR build:

```bash
docker run -d -p 8888:8080 nocodb/nocodb-timely:0.258.2-pr-9903-20241127-1040
```

**Comment by Omasumasu** [comment]
> @Omasumasu It was not related to column name, the real issue happened because it's trying to extract column details from wrong table. We made a fix here - #9903
> 
> You can verify the fix by using following PR build:
> 
> ```shell
> docker run -d -p 8888:8080 nocodb/nocodb-timely:0.258.2-pr-9903-20241127-1040
> ```

@pranavxc 
It fixed!
Thank you so much for the very quick fix ❤️
